packages/core/admin/admin/src/content-manager/components/BlocksEditor/BlocksInput/index.js

```diff
@@ -82,18 +82,36 @@ const BlocksInput = ({ disabled, placeholder }) => {
     }
   };
 
+  /**
+   *  scrollSelectionIntoView : Slate's default method to scroll a DOM selection into the view,
+   *  thats shifting layout for us when there is a overflowY:scroll on the viewport.
+   *  We are overriding it to check if the selection is not fully within the visible area of the editor,
+   *  we use scrollIntoView to scroll it into view smoothly
+   */
+  const handleScrollSelectionIntoView = () => {
+    const domRange = ReactEditor.toDOMRange(editor, editor.selection);
+    const domRect = domRange.getBoundingClientRect();
+    const editorRect = document.getElementById('slate-input').getBoundingClientRect();
```

Comment:
Would it possible to use a ref instead of getElementById?

Reply:
no actually, I tried it but Editable doesnt support ref, I always get null value.
 https://github.com/strapi/strapi/blob/main/packages/core/admin/admin/src/content-manager/components/BlocksEditor/index.js#L120